cmd/overlay/cache.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/provider"
+)
+
+type cacheConfig struct {
+	NodesPath   string `help:"the path to a JSON file containing an object with IP keys and nodeID values" default:"$CONFDIR/nodes.json"`
+	DatabaseURL string `help:"the database connection string to use" default:"bolt://$CONFDIR/overlay.db"`
+}
+type cacheFunc func(*overlay.Cache) error
+type cacheInjector struct {
+	cacheConfig
+	c cacheFunc
+}
+
+func (c cacheInjector) Run(ctx context.Context, server *provider.Provider) (err error) {
```

Comment:
instead of this, could you just have a function that creates an overlay.Cache? instead of an injector and a callback function, a constructor that just returns the thing you want?